### DISCUSSION TITLE
Auto fan curve doesnt work as described

### DISCUSSION THREAD

**Discussion by Popullarmmax** [Rem0o/FanControl.Releases#2425] [Q&A]
During idle, up to load, it will act like a regular linear fan curve. - this part works as described
In the load zone, defined by the Load Temperature and Deadband parameters, it will use a feedback loop to look at the temperature trend and increase or decrease the speed accordingly. If the temperature trend is slowly decreasing, it will decrease the speed little by little until an equilibrium state is found, finding the minimum fan speed required to hold a steady load temperature. - this part isnt. my setings are that its suposed to hold 80c, step 5, dead 3, response 2. what hapens when the load starts the rpm goes up gradualy as it should but when it hits 80c the fan cools it down to 77c because its running at 100% after that the rpm drops to minimum procent and then stays there untill the temperature again hits 80c when the fan ramps up again essentialy making it very anoying. can i fix this somehow? is there any more information needed? i just want my gpu be on the quietest rpm and below 80c when rendering and gaming and this is the only app that alows temperature dependent rpm control... if i could set temperature/rpm in the graph i think that would work too. i wish it would be as simple as saying max temp is 80c use 100% of the fan if necesary and then the software would simply lower the rpm till the temperature would setle. and when the load drops the rpm should drop with the temperature esentialy using the least rpm possible to contain the temperature depending on the load. the fan curve doesnt work for me since the gpu behaves diferently when drawing 140w and 220w and the fan curve can be way less or more for same noise levels. thank you
![image](https://github.com/Rem0o/FanControl.Releases/assets/165744284/024d52a9-4201-4b10-b9f3-d2b757d95e6e)

**Comment by Rem0o** [comment]
Increase the deadband to like 7-8 degree, decrease step de 0.5%, increase response time to like 3-4 seconds.

Auto works best on constant load scenarios.  I would advise using a simple graph in your use case instead.

**Reply by Popullarmmax** [reply]
the thing with the graph is that i cannot set a temperature curve im setting a fan curve which means that when i have a 220w load the curve is acurate and optimised as i manualy configured it but as soon as i get a 100-150w load the fans are spining faster than necesary...thats when i got an idea of trying to just target 80c with minimal speeds... i tried what you wrote at the start of the load the fan jumps between 50 and 80% and in around 5-10 min settles at 57% which is interesting to me...at work we have a pump that automaticaly adjust the speed if the requiered throughput isnt met and on one mashine its like here it jumps up and down but stabalises sooner or later but on the other the pump is more sensitive and reacts almost imediately to change it also looks more radic and overshoots at the start but thats just a second or two of calculation that can be dampened here...maybe just a faster loop back is needed with no history just adjustig parameters acording to the temperature and fan speed in this exact moment and nothing else trying to hit target temperature. also noticed that i cannot choose gpu hotspot temperature but only die temp which is fine its just that at 220w the hotspot is just a tad higher so the gpu starts downclocking but at 140w it doesnt do that so seting an ofset doesnt work since the diference ranges from 1-13c depending on the load. sory if im being anoying but honestly i think this idea is superior than using just a standard fan curve and it could improve cooling and noise for everyone. it can be as simple as seting a target temp and everything else would be automatic and personaly i think this is the only program capable of achieving this and im prepared to donate money if thats whats necesary to achieve this...i would try to program it myself but i know litelary nothing about code thats why im turning here thank you for your help this is an amaying piece of software that is a joy to use
